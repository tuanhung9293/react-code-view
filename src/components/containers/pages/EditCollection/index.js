import React from 'react'
import DocumentTitle from 'components/html/DocumentTitle'
import {connect} from 'react-redux'
import {withRouter} from 'react-router-dom'
import Modal from 'components/ui/modals/Modal'
import ModalHeader from 'components/ui/modals/ModalHeader'
import CollectionForm from '../../shared/CollectionForm'
import {editCollection} from 'state/actions/collectionsAction'
import styles from './EditCollection.module.css'

class EditCollection extends React.Component {
  render() {
    const {collectionData, history} = this.props
    return (
      <Modal>
        <div className={styles.base}>
          <ModalHeader
            heading="Edit Collection"
            closeTo="/collection"
          />
          <div className={styles.form}>
            <DocumentTitle title="Edit Collection" />
            <CollectionForm
              name="editCollection"
              initialValues={collectionData}
              submitAction={editCollection}
              redirectAction={() => history.push('/collection')}
            />
          </div>
        </div>
      </Modal>
    )
  }
}

function mapStateToProps({collections}, {match}) {
  return {
    collectionData: collections.data.find(item => item.nasaId === match.params.collectionId) || {}
  }
}

export default withRouter(connect(mapStateToProps, null)(EditCollection))
