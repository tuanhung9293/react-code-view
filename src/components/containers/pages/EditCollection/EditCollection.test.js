import React from 'react'
import {shallow} from 'enzyme'
import EditCollection from './'

describe('renders', () => {
  it('renders EditCollection connected', () => {
    expect(shallow(
      <EditCollection />
    )).toMatchSnapshot()
  })
})

