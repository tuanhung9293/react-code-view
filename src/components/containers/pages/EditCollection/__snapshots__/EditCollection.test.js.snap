// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders EditCollection connected 1`] = `
<Route>
  <Component />
</Route>
`;
